orbit/pkg/setup_experience/setup_experience.go

```diff
@@ -13,28 +22,227 @@ type Client interface {
 // SetupExperiencer is the type that manages the Fleet setup experience flow during macOS Setup
 // Assistant. It uses swiftDialog as a UI for showing the status of software installations and
 // script execution that are configured to run before the user has full access to the device.
+// If the setup experience is supposed to run, it will launch a single swiftDialog instance and then
+// update that instance based on the results from the /orbit/setup_experience/status endpoint.
 type SetupExperiencer struct {
 	OrbitClient Client
+	closeChan   chan struct{}
+	rootDirPath string
+	// Note: this object is not safe for concurrent use. Since the SetupExperiencer is a singleton,
+	// its Run method is called within a WaitGroup,
+	// and no other parts of Orbit need access to this field (or any other parts of the
+	// SetupExperiencer), it's OK to not protect this with a lock.
+	sd      *swiftdialog.SwiftDialog
+	started bool
 }
 
-func NewSetupExperiencer(client Client) *SetupExperiencer {
-	return &SetupExperiencer{OrbitClient: client}
+func NewSetupExperiencer(client Client, rootDirPath string) *SetupExperiencer {
+	return &SetupExperiencer{
+		OrbitClient: client,
+		closeChan:   make(chan struct{}),
+		rootDirPath: rootDirPath,
+	}
 }
 
 func (s *SetupExperiencer) Run(oc *fleet.OrbitConfig) error {
 	// We should only launch swiftDialog if we get the notification from Fleet.
+	_, binaryPath, _ := update.LocalTargetPaths(
+		s.rootDirPath,
+		"swiftDialog",
+		update.SwiftDialogMacOSTarget,
+	)
+
+	if _, err := os.Stat(binaryPath); err != nil {
+		return nil
+	}
+
 	if !oc.Notifications.RunSetupExperience {
 		log.Debug().Msg("skipping setup experience")
 		return nil
 	}
 
```

Comment:
I just put up a branch that adds a method to check if swiftdialog has been closed, should make some of these checks easier